### PR TITLE
Improve values schema

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Works towards #135 

This PR defines all the properties for opentelemetry-collector's values.schema.yaml.  Actual validations to follow.

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
The amount of changes kinda got away from me.  Let me know what you think and if I should go an add actual validations to this PR as well

File: charts/opentelemetry-collector/values.schema.json [hunk 1]
```diff
@@ -2,30 +2,446 @@
...
+            "Never"
+          ]
+        }
+      }
+    },
+    "imagePullSecrets": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
```
Comment (dmitryax) [inline charts/opentelemetry-collector/values.schema.json:44]: It's not a list of strings in this helm chart. Can be changed later, but currently it's a list of objects like {name: myregistrykey}

File: charts/opentelemetry-collector/values.schema.json [hunk 2]
```diff
@@ -2,30 +2,446 @@
...
+        "annotations": {
+          "type": "object"
+        },
+        "name": {
+          "type": "string"
+        },
+        "rules": {
+          "type": "array",
+          "items": {
+            "type": "string"
```
Comment (dmitryax) [inline charts/opentelemetry-collector/values.schema.json:96]: rule is an object

**Comment by dmitryax** [comment]
> Let me know what you think and if I should go an add actual validations to this PR as well or if I should make it smaller instead.

What actual validation do you mean? Isn't it enforced already?

File: charts/opentelemetry-collector/values.schema.json (hunk 1 above)
Comment (TylerHelmuth) [reply charts/opentelemetry-collector/values.schema.json]: Updated type to be object.

File: charts/opentelemetry-collector/values.schema.json (hunk 2 above)
Comment (TylerHelmuth) [reply charts/opentelemetry-collector/values.schema.json]: Updated type to be object.

**Comment by TylerHelmuth** [comment]
Looking through the only other validation I could come up with was something fancy around standaloneCollector.enabled and podDisruptionBudget/autoscaling, but I think all that matters is that enabled is required by the properties.  The actual value of enabled doesn't matter. So I think this PR gets the schema into a good state to validate the current values.yaml.

**Comment by dmitryax** [comment]
> Looking through the only other validation I could come up with was something fancy around standaloneCollector.enabled and podDisruptionBudget/autoscaling, but I think all that matters is that enabled is required by the properties. The actual value of enabled doesn't matter. So I think this PR gets the schema into a good state to validate the current values.yaml.

I don't the fancy things are that important to have. My idea is that we need to define all the properties with `"additionalProperties": false` in order to make sure that any replacements of the existing configuration parameters blocks the chart upgrade. It make any breaking change at least not silently breaking.

**Review by dmitryax — APPROVED** [review]
LGTM

**Pull request merged**